src/main/java/org/spongepowered/api/service/economy/Account.java

```diff
@@ -0,0 +1,178 @@
+	 * 
+	 * @return a {@link BigDecimal} value.
+	 */
+	BigDecimal getBalance(Currency currency);
+	
+	/**
+	 * Returns a {@link BigDecimal} representative of the balance stored within this
+	 * {@link Account} for the {@link Currency} given.
+	 * 
+	 * <p>The default result when the account does not have a balance of the given
+	 * {@link Currency} should be {@link BigDecimal#ZERO}.</p>
+	 * 
+	 * @param a {@link Currency} to check the balance on.
+	 * 
+	 * @return a {@link BigDecimal} value.
+	 */
+	BigDecimal getBalance(Currency currency, Set<Context> contexts);
+
+	/**
+	 * Returns a {@link Map} of all currently set balances the account holds.
+	 * 
+	 * <p>{@link Currency} amounts which are 0 may or may not be included in the
+	 * returned mapping.</p>
+	 * 
+	 * <p>Changes to the returned {@link Map} will not be reflected in the underlying
+	 * {@link Account}. See {@link #setBalance(BigDecimal, Currency)} to set values.</p>
+	 * 
+	 * @return {@link Map} of {@link Currency} to {@link BigDecimal} amounts that this
+	 *         account holds.
+	 */
+	Map<Currency, BigDecimal> getBalances();
+
+	/**
+	 * Sets the balance stored by this {@link Account} to the given amount for
+	 * the {@link Currency} specified.
+	 * 
+	 * <p>Negative balances may or may not be supported depending on
+	 * the {@link Currency} specified. See {@link Currency#canGoNegative()}.</p>
+	 * 
+	 * @param currency the {@link Currency} to be used.
+	 * @param amount a {@link BigDecimal} value representing the new balance
+	 *        for the {@link Currency} given.
+	 * @param a {@link Currency} 
+	 */
+	TransactionResult setBalance( Currency currency, BigDecimal amount);
+
+	/**
+	 * Adds the given amount of the specified {@link Currency} to this {@link Account}.
+	 * 
+	 * @param currency the {@link Currency} to be used.
+	 * @param amount a {@link BigDecimal} representative of the amount to give to the target
+	 *        Account
+	 * 
+	 * @return a {@link TransactionResult} representative of the effects of the
+	 *         operation
+	 */
+	TransactionResult deposit(Currency currency, BigDecimal amount);
+	
+	/**
+	 * Adds the given amount of the {@link Currency} to this {@link Account} 
+	 * in the given contexts.
+	 * 
+	 * <p>If the account is not in the given {@link Context} the result should be
+	 * a failure</p>
+	 * 
+	 * @param amount a Number representative of the amount to give to the target
+	 *        Account
+	 * 
+	 * @return a {@link TransactionResult} representative of the effects of the
+	 *         operation
+	 */
+	TransactionResult deposit(Currency currency, Number amount, Set<Context> contexts);
+
+	/**
+	 * Removes the given amount of the {@link Currency} from this {@link Account}.
+	 * 
+	 * @param amount a Number representing the amount to remove from the account
+	 * 
+	 * @return a {@link TransactionResult} representative of the effects of the
+	 *         operation
+	 */
+	TransactionResult withdraw(Currency currency, Number amount);
```

Comment:
Why separate methods for deposit and withdraw? You can just "deposit a negative value".